application/namespaces/user/controllers.py

```diff
@@ -0,0 +1,180 @@
+    OrodhaNotFoundError,
+    OrodhaForbiddenError,
+)
+
+APPCONFIG = obtain_config()
+
+
+def _add_user_to_database(user_args: dict) -> User:
+    """
+    Helper function which takes user_args and creates a mongo user.
+
+    Args:
+        user_args(dict): A dictionary of our user's data such as
+            username and email.
+
+    Returns:
+        User: The user Document blueprint that is defined in models.py,
+            filled with data from user_args
+    """
+    return User(**user_args).save()
+
+
+def _create_keycloak_client() -> orodha_keycloak.OrodhaKeycloakClient:
+    """
+    Helper function which creates our keycloak client from config data
+    for use in our main controller functions.
+
+    Returns:
+        OrodhaKeycloakClient: A facade client used by Orodha in order to make interactions with
+            keycloak more uniform for the service.
+    """
+    return orodha_keycloak.OrodhaKeycloakClient(
+        server_url=APPCONFIG["keycloak_config"]["keycloak_server_url"],
+        realm_name=APPCONFIG["keycloak_config"]["keycloak_realm_name"],
+        client_id=APPCONFIG["keycloak_config"]["keycloak_client_id"],
+        client_secret_key=APPCONFIG["keycloak_config"]["keycloak_client_secret_key"],
+    )
+
+
+def get_user(token: str, request_user_mongo_id: str) -> User:
+    """
+    Function that takes a token and finds the user if any in our database associated with it.
+
+    Args:
+        token(str): The token passed from the UserApi route which helps us identify user auth level.
+        request_user_mongo_id(str): The user id that is associated with the user that we want to get.
+
+    Returns:
+        user(document): A mongoengine document object containing the data of the requested user.
+
+    Raises:
+        OrodhaBadIdError: If there is a problem with the KeycloakGet method, typically because of
+            a bad id passed in.
+        OrodhaNotFoundError: If there are either no matching objects in the mongo db, or if there
+            are more than one object that match the same id.
+    """
+    try:
+        keycloak_client = _create_keycloak_client()
+        keycloak_id_from_token = keycloak_client.get_user(token=token).get("id")
+
+        user = User.objects.get(keycloak_id=keycloak_id_from_token)
+
+        if str(user.id) != request_user_mongo_id:
+            raise OrodhaForbiddenError(
+                message="You don't have permission to access this resource"
+            )
+
+        return user
+
+    except KeycloakGetError as err:
+        raise OrodhaBadIdError(
+            f"Unable to find keycloak user with token {token}: {err}"
+        )
+    except (MultipleObjectsReturned, DoesNotExist) as err:
+        raise OrodhaNotFoundError(
+            f"Unable to find unique user with userId {request_user_mongo_id}: {err}"
+        )
+
+
+def post_user(payload: dict) -> User:
+    """
+    Funtion which creates a new keycloak user and a new database entry from user_data.
```

Comment:
@Edison-Stuart Check the docstring spelling

Reply:
Right I missed that one, Thank you!